src/multi_vector_simulator/D1_model_components.py

```diff
@@ -336,20 +336,19 @@ def transformer_constant_efficiency_fix(model, dict_asset, **kwargs):
     ):
         if isinstance(dict_asset[INPUT_BUS_NAME], list):
             inputs = {}
-            index = 0
             for bus in dict_asset[INPUT_BUS_NAME]:
-                variable_costs = dict_asset[DISPATCH_PRICE][VALUE][index]
-                inputs[kwargs[OEMOF_BUSSES][bus]] = solph.Flow(
+                inputs[kwargs[OEMOF_BUSSES][bus]] = solph.Flow()
+            outputs = {
+                kwargs[OEMOF_BUSSES][dict_asset[OUTPUT_BUS_NAME]]: solph.Flow(
                     nominal_value=dict_asset[INSTALLED_CAP][VALUE],
-                    variable_costs=variable_costs,
-                )
-                index += 1
-            outputs = {kwargs[OEMOF_BUSSES][dict_asset[OUTPUT_BUS_NAME]]: solph.Flow()}
-            efficiencies = {}
-            for i in range(len(dict_asset[EFFICIENCY][VALUE])):
-                efficiencies[kwargs[OEMOF_BUSSES][dict_asset[INPUT_BUS_NAME][i]]] = (
-                    1 / dict_asset[EFFICIENCY][VALUE][i]
+                    variable_costs=dict_asset[DISPATCH_PRICE][VALUE],
                 )
+            }
+            efficiencies = {
+                kwargs[OEMOF_BUSSES][dict_asset[OUTPUT_BUS_NAME]]: dict_asset[
+                    EFFICIENCY
+                ][VALUE]
+            }
 
         else:
             inputs = {kwargs[OEMOF_BUSSES][dict_asset[INPUT_BUS_NAME]]: solph.Flow()}
```

Comment:
Isnt that now a duplication of the same code that is used for transformers that are fix/to be optimized? You could summarize it in a single processing function.

Reply:
No they are different (for example investment object only needed for optimzation), as well as for sources fix/optimize etc. 
If we want to process them within one function each I would to that in a new PR - but for now I don't see an urgency.

Reply:
Ah, of course the last part is different, but the processing of `efficiency`, `variable_costs` and `outputs`  seems to be identical and could be wrapped in a function. You are right though, that can be done in a different PR. Do your tests cover everything already? Then we do not need to write an issue for this.

Reply:
True, they could be processed within one function (probably accounts for sources, sinks, etc., too). Main difference: the oemof.solph.Flow(). I'll write an issue.

Reply:
True, they could be processed within one function (probably accounts for sources, sinks, etc., too). Main difference: the oemof.solph.Flow(). I'll write an issue.